image/store/layer_store.go

```diff
@@ -54,6 +54,24 @@ func (ls *layerStore) RegisterLayerIfNotPresent(closer io.ReadCloser, id LayerID
 	return nil
 }
 
+func (ls *layerStore) Delete(id LayerID) error {
+	layer := ls.Get(id)
+	if layer == nil {
+		logger.Debug("layer %s is already deleted", id)
+		return nil
+	}
+
+	filePath := filepath.Join(common.DefaultLayerDir, digest.Digest(id).Hex())
+	utils.CleanDir(filePath)
```

Comment:
I guess we may still need catch the err, if cleanDir failed, the layer stilll exitss